packages/security_center/lib/widgets/passphrase_widgets.dart

```diff
@@ -0,0 +1,239 @@
+        ),
+      ]),
+      onChanged: (value) {
+        notifier.newPass = value;
+      },
+    );
+  }
+}
+
+class ConfirmPassphraseFormField extends ConsumerStatefulWidget {
+  const ConfirmPassphraseFormField({
+    required this.authMode,
+    this.fieldWidth,
+    super.key,
+  });
+
+  final double? fieldWidth;
+  final AuthMode authMode;
+
+  @override
+  ConsumerState<ConfirmPassphraseFormField> createState() =>
+      _ConfirmPassphraseFormFieldState();
+}
+
+class _ConfirmPassphraseFormFieldState
+    extends ConsumerState<ConfirmPassphraseFormField> {
+  final _controller = TextEditingController();
+
+  @override
+  void initState() {
+    final model = ref.read(changeAuthDialogModelProvider);
+    assert(widget.authMode != AuthMode.none);
+
+    super.initState();
+    _controller.value = _controller.value.copyWith(text: model.confirmPass);
+    if (widget.authMode == AuthMode.pin) {
+      _controller.addListener(() => _filterDigits(_controller));
+    }
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final model = ref.watch(changeAuthDialogModelProvider);
+    final notifier = ref.watch(changeAuthDialogModelProvider.notifier);
+    final lang = AppLocalizations.of(context);
+    final isDisabled = model.dialogState is ChangeAuthDialogStateSuccess;
+
+    return ValidatedFormField(
+      controller: _controller,
+      fieldWidth: widget.fieldWidth,
+      labelText: widget.authMode.localizedConfirmHint(lang),
+      obscureText: !model.showPassphrase,
+      suffixIcon: const _SecurityKeyShowButton(),
+      enabled: !isDisabled,
+      validator: EqualValidator(
+        model.newPass,
+        errorText: widget.authMode.localizedConfirmError(lang),
+      ),
+      onChanged: (value) {
+        notifier.confirmPass = value;
+      },
+    );
+  }
+}
+
+void _filterDigits(TextEditingController controller) {
+  final filteredInput = controller.text.replaceAll(RegExp('[^0-9]'), '');
+  final newSelection = TextSelection(
+    baseOffset: min(controller.selection.baseOffset, filteredInput.length),
+    extentOffset: min(controller.selection.extentOffset, filteredInput.length),
+  );
+  controller.value = controller.value.copyWith(
+    text: filteredInput,
+    selection: newSelection,
+  );
+}
+
+class _SecurityKeyShowButton extends ConsumerWidget {
+  const _SecurityKeyShowButton();
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    final theme = Theme.of(context);
+    final lang = AppLocalizations.of(context);
+    final model = ref.watch(changeAuthDialogModelProvider);
+    final notifier = ref.read(changeAuthDialogModelProvider.notifier);
+    final showSecurityKey = model.showPassphrase;
+    final isDisabled = model.dialogState is ChangeAuthDialogStateSuccess;
+
+    return Padding(
+      padding: const EdgeInsets.all(1.0),
+      child: TextButton.icon(
+        icon: Icon(
+          showSecurityKey ? YaruIcons.eye : YaruIcons.hide,
```

Comment:
```suggestion
          showSecurityKey ? YaruIcons.hide : YaruIcons.eye,
```
Juan pointed out that those are reversed - the icon is supposed to indicate the action, not the state. I've fixed this in the open installer PR including the entropy checks